src/core/src/pass/serialize.cpp

```diff
@@ -1146,13 +1146,12 @@ void ngfunction_2_ir(pugi::xml_node& netXml,
     }
 }
 
-std::string valid_xml_path(const std::string& path) {
-    OPENVINO_ASSERT(path.length() > 4, "Path for xml file is too short: \"" + path + "\"");
-
+std::string convert_to_valid_xml_path(const std::string& path) {
     const char* const extension = ".xml";
     const bool has_xml_extension = path.rfind(extension) == path.size() - std::strlen(extension);
-    OPENVINO_ASSERT(has_xml_extension,
-                    "Path for xml file doesn't contains file name with 'xml' extension: \"" + path + "\"");
+    if (!has_xml_extension) {
+        return ov::util::path_join({path, "model.xml"});
```

Comment:
should we check that `path` is a directory before composing a new path?

Also, I think we need to update documentation for `save_model` function in this PR just to make sure we are aligned here.

Reply:
Checked this case when we pass invalid path, `ov::util::create_directory_recursive()` fails to create directory in this case  here:
https://github.com/openvinotoolkit/openvino/blob/e8b6e1742906c60741792dfa5b75008033029192/src/common/util/src/file_util.cpp#L406
```
Error example:
RuntimeError: Couldn't create directory [./kk:], err=Invalid argument)
```
This looks enough. 

Regarding documentation. I can't find any articles about `save_model()`, only examples in Notebooks. Should I add info in docs about `save_model()`?

Reply:
Am I correct that if the path does not contain `.xml` extension, it will be always interpreted as a directory? Should we say about this in the documentation because I thought the extension is added automatically like we have in MO?

Reply:
@popovaan, @slyalin, who is a customer of this feature? This definition of the feature looks strange to me.

Reply:
> Am I correct that if the path does not contain .xml extension, it will be always interpreted as a directory? Should we say about this in the documentation because I thought the extension is added automatically like we have in MO?
Yes, if the path does not contain .xml, it will be always interpreted as a directory. OV tries to create this directory if it does not exist and save model with name model.xml. OVC has currently the same logic.